### ISSUE TITLE
Error: Encrypted content has been tampered with

### ISSUE THREAD

**Issue by anbinh** [himynameisdave/git-labelmaker#32]
I got this error when running the program 

<img width="642" alt="capture d ecran 2016-02-22 a 12 21 17" src="https://cloud.githubusercontent.com/assets/340262/13226311/da0c7516-d95e-11e5-82d2-3ee52f25d071.png">

```
➜ ~/www/veldto.com (master) git-labelmaker
? What is your master password? *********
Failed opening archive: Error: Encrypted content has been tampered with
{ id: 'TOKEN',
  err: 'Failed opening archive: Error: Encrypted content has been tampered with' }
```

**Comment by himynameisdave** [comment]
@anbinh not cool... thanks for reporting, what version of node are you currently running (`node -v`)?

**Comment by jaimeeee** [comment]
The same thing is happening to me, my node version is 5.6.0.

**Comment by kevmoo** [comment]
Ditto, with git-labelmaker 0.7.0

**Comment by himynameisdave** [comment]
~~Tested on node `v5.5.0`and it works, something about `v5.6.0`...~~ Still investigating

**Comment by jaredmichaelwilliams** [comment]
@himynameisdave I'm on `v5.5.0` and I'm getting that error

```
Jared@Jared-Williams-iMac /git/artbinder/ArtbinderMobile [master *]
± % node -v                                                                                                                                                                                                                                                                                                                                                                                            !39096
v5.5.0

Jared@Jared-Williams-iMac /git/artbinder/ArtbinderMobile [master *]
± % git-labelmaker                                                                                                                                                                                                                                                                                                                                                                                     !39097
? What is your master password? ****************************************
Failed opening archive: Error: Encrypted content has been tampered with
{ id: 'TOKEN',
  err: 'Failed opening archive: Error: Encrypted content has been tampered with' }
```

**Comment by himynameisdave** [comment]
Thanks still looking into this...

**Comment by alchenist** [comment]
node `v5.4.1` and also getting this error. Got prompted for master password without passing in token or an initial password.

```
C:\Users\AC\Projects\test>node -v
v5.4.1

C:\Users\AC\Projects\test>git-labelmaker
? What is your master password?
Failed opening archive: Error: Encrypted content has been tampered with
{ id: 'TOKEN',
  err: 'Failed opening archive: Error: Encrypted content has been tampered with' }
```

**Comment by jaredmichaelwilliams** [comment]
@himynameisdave Hey, so i reverted back to 

```
commit 2be40beaacc90a6e4b11d629da0704a0ac2e23ca
Author: dlunny <dlunny@parka.ca>
Date:   Sun Feb 21 21:40:59 2016 -0500

    Readme cleanup
```

and rebuilt and now its working on 5.6.0

**Comment by himynameisdave** [comment]
@Jameskmonger any ideas what's going on here?

**Comment by Jameskmonger** [comment]
@himynameisdave

I believe that error is actually the "invalid password" error... Are you guys putting in the wrong password?

**Comment by jaredmichaelwilliams** [comment]
@Jameskmonger this is the first time i tried to use it, i never even got to put in a token or password.

**Comment by himynameisdave** [comment]
Yeah @Jameskmonger you're right about it being an invalid password error, but the real problem is that it's expecting a password before the user has even set it, like @jaredmichaelwilliams is saying... hmmmm... **(╯°□°)╯︵ ┻━┻**

**Comment by Jameskmonger** [comment]
@jaredmichaelwilliams ah okay, in that case I think I might know what the issue is. @himynameisdave could you assign this to me please?

**Comment by himynameisdave** [comment]
@Jameskmonger it's all you man! I'm still l:eyes:king into this though...

**Comment by Jameskmonger** [comment]
This is now ready for code review in #32.

@jaredmichaelwilliams if you wouldn't mind, could you clone the branch `32-fix` and run `sudo npm install -g ./` in the directory? This will install the version 0.7.1-alpha which should fix this issue. Please let me know.

**Comment by Jameskmonger** [comment]
Issue still persisting in 0.7.1-alpha, node 5.6.0

**Comment by himynameisdave** [comment]
Damn. Can we isolate it to just that version of node? If so, then maybe [we can see if there was some change](https://nodejs.org/en/blog/release/v5.6.0/) that might have caused this...?

Maybe it's an issue with [`buttercup`](https://github.com/buttercup-pw/buttercup-core)?

**Comment by Jameskmonger** [comment]
Issue is also appearing on node 5.0.0 @himynameisdave

**Comment by Jameskmonger** [comment]
Oddly enough @himynameisdave the issue doesn't seem to appear when running locally with either of the following two commands:

`sudo npm install -g ./ && git-labelmaker`

(install the local version of the package and run the global `git-labelmaker` command)

`npm install && node bin/index.js`

(install all dependencies and then run the local `index.js` file)

However the following installation process does introduce the error:

`sudo npm install -g git-labelmaker@0.7.1-alpha && git-labelmaker`

This makes me think there is possibly an issue in the publishing process between what is being published and what is being downloaded through npm...

**Comment by Jameskmonger** [comment]
@himynameisdave I've done a fresh clone of the repo's master branch as it is at the moment, and then I published it to npm under the name `git-labelmaker-32fix`. I then ran the following commands:

```
$ sudo npm install -g git-labelmaker-32fix
$ git-labelmaker
```

And the command works correctly, detecting that I have not yet chosen a token and prompting me as intended. There appears to be some kind of issue with the release that has been uploaded onto npm as `git-labelmaker`.

If you're willing to add me as a collaborator on npm I will be able to take a look at it in the morning!

Good luck :beers:

**Comment by himynameisdave** [comment]
@Jameskmonger will do, meantime still trying to get to the bottom of this...

**Comment by Jameskmonger** [comment]
No problem @himynameisdave. I've tested `git-labelmaker-32fix` on node `v5.6.0` and `v5.0.0` and it's behaving as expected, while `git-labelmaker@0.7.1-alpha` is encountering the bug on the same versions. It's very odd, not sure how or why that could have happened.

Potentially your local git repo had some kind of issue before you ran `npm publish`? Seems a bit of a stretch, though...

**Comment by alchenist** [comment]
On a fresh install of `git-labelmaker@0.7.0` from npm, noticed that there was an existing `.git-labelmaker.bcup` archive in the `node_modules\git-labelmaker` directory. Deleted this and everything worked fine. Somewhere along the line, an existing credentials archive is being included in the version published to npm.

**Comment by Jameskmonger** [comment]
That makes perfect sense @alchenist!

@himynameisdave, we will need to add `.git-labelmaker.bcup` to `.npmignore` and republish on npm (probably best to publish as 0.7.2)

**Comment by mrbaguettefr** [comment]
Thank you @alchenist,  I executed this command to delete the .git-labelmaker.bcup in the global packages of node and everything worked fine.
`rm -f /usr/local/lib/node_modules/git-labelmaker/.git-labelmaker.bcup`

**Comment by himynameisdave** [comment]
@alchenist @Jameskmonger You guys are awesome, thanks for getting to the bottom of this! :+1: :smile:  :tada: 

I should always create an `.npmignore` file when starting a new project, even if in my mind I don't think I'll need it.

**Comment by Jameskmonger** [comment]
No problem @himynameisdave! I've made a pull request which should solve this.

**Comment by Jameskmonger** [comment]
This is now solved in version `0.7.2`.

Thanks for your help in diagnosing this issue: @anbinh, @jaimeeee, @kevmoo, @jaredmichaelwilliams, @alchenist, @mrbaguette :fireworks: 

You can download the new version with the following command:

```
sudo npm install -g git-labelmaker
```
